components/script/dom/request.rs

```diff
@@ -556,7 +556,7 @@ impl RequestMethods for Request {
 
     // https://fetch.spec.whatwg.org/#dom-request-mode
     fn Mode(&self) -> RequestMode {
-        self.request.borrow().mode.into()
+        self.request.borrow().mode.clone().into()
```

Comment:
why is the clone needed?

Reply:
`RequestMode` used to implement `Copy`, but since I've added the `WebSocket(Vec<String>)` variant, it cannot implement `Copy`, hence this clone.